crates/astria-conductor/src/executor/state.rs

```diff
@@ -1,60 +1,95 @@
-use astria_core::execution::v1alpha2::{
-    Block,
-    CommitmentState,
+use astria_core::{
+    execution::v1alpha2::{
+        Block,
+        CommitmentState,
+        GenesisInfo,
+    },
+    sequencer::v1alpha1::RollupId,
 };
+use celestia_client::celestia_types::Height as CelestiaHeight;
 use sequencer_client::tendermint::block::Height;
 
-// Maps a rollup height to a sequencer heights.
-/// # Panics
-///
-/// Panics if adding the integers overflows. Comet BFT has hopefully migrated
-/// to `uint64` heights by the times this becomes an issue.
-fn map_rollup_height_to_sequencer_height(
-    first_sequencer_height: u32,
-    current_rollup_height: u32,
-) -> Height {
-    first_sequencer_height
-        .checked_add(current_rollup_height)
-        .expect(
-            "should not overflow; if this overflows either the first sequencer height or current \
-             rollup height have been incorrectly set incorrectly set, are in the future, or the \
-             rollup/sequencer have been running for several decades without cometbft ever \
-             receiving an update to uin64 heights",
-        )
-        .into()
+#[derive(Debug)]
+pub(crate) struct State {
+    inner: Option<StateImpl>,
 }
 
 #[derive(Debug)]
-pub(crate) struct State {
+struct StateImpl {
+    genesis_info: GenesisInfo,
     commitment_state: CommitmentState,
 
     next_firm_sequencer_height: Height,
     next_soft_sequencer_height: Height,
-
-    // The sequencer height that contains the first block of the executed-upon rollup.
-    sequencer_height_with_first_rollup_block: u32,
 }
 
 impl State {
-    pub(super) fn new(
-        commitment_state: CommitmentState,
-        sequencer_height_with_first_rollup_block: u32,
-    ) -> Self {
+    pub(super) fn new() -> Self {
+        Self {
+            inner: None,
+        }
+    }
+
+    pub(super) fn init(&mut self, genesis_info: GenesisInfo, commitment_state: CommitmentState) {
+        self.inner
+            .replace(StateImpl::new(genesis_info, commitment_state));
+    }
```

Comment:
should we make sure this is called only once?

Reply:
Replace this with `*self.inner.as_mut().expect("msg") = ...`.